phoenix-core/src/main/java/org/apache/phoenix/compile/ScanRanges.java

```diff
@@ -374,9 +375,19 @@ public Scan intersectScan(Scan scan, final byte[] originalStartKey, final byte[]
         if (scanStopKey.length > 0 && Bytes.compareTo(scanStartKey, scanStopKey) >= 0) { 
             return null; 
         }
+
+        ImmutableBytesWritable a1 = new ImmutableBytesWritable(scanStartKey);
+        ImmutableBytesWritable b1 = new ImmutableBytesWritable(scanStopKey);
+        byte[] a1StartKey = a1.copyBytes();
+        byte[] b1StartKey = b1.copyBytes();
+        newScan.setAttribute(SCAN_ACTUAL_START_ROW, a1StartKey);
+        newScan.setStartRow(a1StartKey);
+        newScan.setStopRow(b1StartKey);
+        /*
         newScan.setAttribute(SCAN_ACTUAL_START_ROW, scanStartKey);
```

Comment:
why do we need to keep the old code here? 
From my point of view, `setStartRow` and `setStopRow` APIs are deprecated, and it will remove soon.

Reply:
I forgot to delete the old code. Using setStartRow and setStopRow are original code which aren't supposed to change in this PR.

Reply:
Resolved.